core/src/wallet/tezos/TezosLikeAccount2.cpp

```diff
@@ -367,8 +367,18 @@ namespace ledger {
                                                 if (!counter) {
                                                     throw make_exception(api::ErrorCode::RUNTIME_ERROR, "Failed to retrieve counter from network.");
                                                 }
-                                                // We should increment current counter
-                                                tx->setCounter(std::make_shared<BigInt>(++(*counter)));
+                                                auto explorer_counter = counter->toInt64();
+                                                auto saved_counter = self->getInternalPreferences()->getLong("counter", 0);
```

Comment:
Not sure we should use internalPreferences database for this. Should we store that, probably an additional column in tezosAccounts is better ?

Reply:
Sure that would be a viable alternative, I first begun by modifying migrate/rollback in migrations but I finally used  internalPreferences because I liked this solution better, I guess it's more a matter of personal preference in the end ?